File: Annexe1/app/src/main/java/com/example/annexe1/Afficher.java
package com.example.annexe1;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Spinner;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Vector;

public class Afficher extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_afficher);

        Spinner spinner = findViewById(R.id.spinner);
        Vector<String> v = new Vector<>();

        Intent intent = getIntent();
        String message = intent.getStringExtra("nouvelleTache");
        if (message != null) {
            v.add(message);
        }

        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, v);
        spinner.setAdapter(arrayAdapter);
    }
}

File: Annexe1/app/src/main/java/com/example/annexe1/Afficher.kt
package com.example.annexe1

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import java.io.BufferedReader
import java.io.FileInputStream
import java.io.InputStreamReader
import java.util.Vector

class Afficher : AppCompatActivity() {

    lateinit var listView: ListView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_afficher)

        listView = findViewById(R.id.liste_memo)
        val arrayAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, lesMemos())
        listView.setAdapter(arrayAdapter)
    }
    fun lesMemos(): Vector<String>
    {
        val v = Vector<String>()
        val fis : FileInputStream = openFileInput("fichier.txt")// append : écrit à la fin du fichier plutot qu'au début
        val isr = InputStreamReader(fis) // traduit en caractères
        val br = BufferedReader(isr) // plus rapide
        br.forEachLine { s -> v.add(s) }
        br.close()
        return v
    }
}
